SCClassLibrary/Common/Control/Server.sc

```diff
@@ -380,41 +421,47 @@ Server {
 	}
 
 	newBusAllocators {
-		var numControl, numAudio;
-		var controlBusOffset, audioBusOffset;
-		var offset = this.calcOffset;
-		var n = options.maxLogins ? 1;
+		var numControlPerClient, numAudioPerClient;
+		var controlReservedOffset, controlBusClientOffset;
+		var audioReservedOffset = 0, audioBusClientOffset;
 
-		numControl = options.numControlBusChannels div: n;
-		numAudio = options.numPrivateAudioBusChannels div: n;
+		numControlPerClient = options.numControlBusChannels div: this.numClients;
+		numAudioPerClient = options.numAudioBusChannels div: this.numClients;
 
-		controlBusOffset = numControl * offset + options.reservedNumControlBusChannels;
-		audioBusOffset = options.firstPrivateBus + (numAudio * offset) + options.reservedNumAudioBusChannels;
+		controlReservedOffset = options.reservedNumControlBusChannels;
+		controlBusClientOffset = numControlPerClient * clientID;
 
-		controlBusAllocator =
-		ContiguousBlockAllocator.new(numControl, controlBusOffset);
-
-		audioBusAllocator =
-		ContiguousBlockAllocator.new(numAudio, audioBusOffset);
+		// only reserve hardware output chans on clientID 0
```

Comment:
are we all sure that we want to have this asymmetry?

Reply:
from what I understand, clientID 0 is the local machine; how could we reserve hardware outputs on remote machines?

Reply:
any other clientID might be on the local machine, if you have several clients on one machine.

Reply:
> any other clientID might be on the local machine

but there's no way of knowing for sure? i don't understand what the implied alternative is

Reply:
in other words how would one make it "symmetric"

Reply:
symmetric means to take the hardware IO offset off the whole range, and divide the rest.
I checked, Julian is right - 3.7.2 and 3.8.0 did this, so I will go back to that:
```
newNodeAllocators { 
...
		numControl = options.numControlBusChannels div: n;
		numAudio = options.numPrivateAudioBusChannels div: n;
		controlBusOffset = numControl * offset;
		audioBusOffset = options.firstPrivateBus + (numAudio * offset);
...
}```
thanks for spotting that, Julian!

Reply:
excellent! good spot :)

Reply:
and done as well.